loader/wsi.c

```diff
@@ -449,6 +449,16 @@ LOADER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkCreateSwapchainKHR(VkDevice devic
                    "vkCreateSwapchainKHR: Invalid device [VUID-vkCreateSwapchainKHR-device-parameter]");
         abort(); /* Intentionally fail so user can correct issue. */
     }
+    // Since an application can get vkCreateSwapchainKHR from vkGetInstanceProcAddr, we need to make sure the extension
+    // was enabled on the device, as if it wasn't, disp->CreateSwapchainKHR will be NULL
+    struct loader_device *dev = *((struct loader_device **)device);
+    if (NULL == dev || !dev->extensions.khr_swapchain_enabled) {
```

Comment:
I would keep NULL device separate and abort because that's something really bad.  The extension not being enabled is not so bad.

Reply:
I think the NULL == dev check is redundant because the above check for the dispatch table should always catch this situation.

Reply:
Turns out I don't need to check if the extension is enabled, just if the function pointer is NULL or not - since it'll be NULL when the user didn't enable the swapchain extension, as disp->vkCreateSwapchainKHR is now correctly filled out with either the terminator_CreateSwapchainKHR or NULL.